cpp/src/structs/utilities.cpp

```diff
@@ -207,73 +210,124 @@ flattened_table flatten_nested_columns(table_view const& input,
   return table_flattener{input, column_order, null_precedence, nullability}();
 }
 
-void superimpose_nulls(bitmask_type const* null_mask,
-                       size_type null_count,
-                       column& input,
-                       rmm::cuda_stream_view stream,
-                       rmm::mr::device_memory_resource* mr)
+namespace {
+
+/**
+ * @brief Conservatively check if the input column needs to be sanitized (i.e., if it may contain
+ * non-empty nulls).
+ *
+ * For performance reason, no non-empty null will actually be checked. Instead, this function
+ * performs conservative checking to see if the input column contains any lists or strings column
+ * that is nulllable. As such, it may return false positive answer.
+ *
+ * @param col The input column to check
+ */
+auto may_need_sanitize(column_view const& col)
 {
-  if (input.type().id() == cudf::type_id::EMPTY) {
+  switch (col.type().id()) {
```

Comment:
Use type-dispatcher instead of switch statement on types.

Reply:
Sure.